modules/openxAnalyticsAdapter.js

```diff
@@ -655,6 +664,80 @@ function pushEvent(eventType, args, auctionId) {
   }
 }
 
+function updateLoadedAdSlotsInfo(auctionId, adUnitCode, adPosition) {
+
+  if(auctionId && adUnitCode) {
+    if(!loadedAdSlots[auctionId]){
+      loadedAdSlots[auctionId] = {};
+    }
+    loadedAdSlots[auctionId][adUnitCode] = {};
+    if(adPosition) {
+      loadedAdSlots[auctionId][adUnitCode] = { adPosition: adPosition };
+    }
+  } else {
+    utils.logWarn("OX: Couldn't update loadedAdSlots information.");
+  }
+}
+
+function getLoadedAdUnitCodes(auctionId) {
+
+  return (!auctionId || !loadedAdSlots[auctionId] || typeof loadedAdSlots[auctionId] !== 'object')
+    ? [] : Object.keys(loadedAdSlots[auctionId]);
+}
+
+function pushAdPositionData(auctionId) {
+
+  if(auctionId && eventStack[auctionId] && eventStack[auctionId].events) {
+
+    let adUnitPositionMap = loadedAdSlots[auctionId];
+    if(adUnitPositionMap && JSON.stringify(adUnitPositionMap) !== "{}") {
+
+      eventStack[auctionId].events.filter(function(event) {
+        return event.eventType === auctionEndConst;
+      }).forEach(function (auctionEndEvent) {
+
+        if(auctionEndEvent.args && auctionEndEvent.args.adUnits) {
+          auctionEndEvent.args.adUnits.forEach(function (adUnitInfo) {
+            if(adUnitPositionMap[adUnitInfo.code] && adUnitPositionMap[adUnitInfo.code]["adPosition"]) {
+              adUnitInfo["adPosition"] = adUnitPositionMap[adUnitInfo.code]["adPosition"];
+            } else {
+              adUnitInfo["adPosition"] = "";
+            }
+          })
+        }
+      });
+    }
+  }
+}
+
+function getAdPositionByElementId(elementId) {
+
+  let elem = document.querySelector("#" + elementId);
+  let adPosition;
+  if (elem) {
+    let bounding = elem.getBoundingClientRect();
+    if (bounding) {
+      let windowWidth = (window.innerWidth || document.documentElement.clientWidth);
+      let windowHeight = (window.innerHeight || document.documentElement.clientHeight);
+
+      // intersection coordinates
+      let left = Math.max(0, bounding.left);
+      let right = Math.min(windowWidth, bounding.right);
+      let bottom = Math.min(windowHeight, bounding.bottom);
+      let top = Math.max(0, bounding.top);
+
+      let intersectionArea = (right - left) * (bottom - top);
```

Comment:
This may give you a false-positive in situations where the adslot element is outside the screen.   Since this is hard to unit test, you can cheat by exposing this function and passing in the values explicitly.

For example:
Window is: 100 width and 100 height
Ad slot bounding box is 100 pixels below the window and 100 pixels to the right of the window.
(top, right , bottom, left): 200, 300, 300, 200
```
left = 200
right = 100
bottom = 100
right = 100

intersection = 10000
adslot area = 10000

Reply:
@jimee02 Updated the logic to handle such false positives. Please have a look.